### DISCUSSION TITLE
v1.1.5

### DISCUSSION THREAD

**Discussion by jp7677** [avisi-cloud/structurizr-site-generatr#342] [Announcements]
## What's Changed

* Add the ability to set the target for links in the SVG file. by @qtzar in https://github.com/avisi-cloud/structurizr-site-generatr/pull/283
* Support links between ADR's by @dirkgroot in https://github.com/avisi-cloud/structurizr-site-generatr/pull/284
* Add custom style sheet support by @qtzar in https://github.com/avisi-cloud/structurizr-site-generatr/pull/315
* Add support for nested groups in menu by @qtzar in https://github.com/avisi-cloud/structurizr-site-generatr/pull/180
* Add support for documentation sections on containers by @jp7677 in https://github.com/avisi-cloud/structurizr-site-generatr/pull/325
* Add pan and zoom for diagrams by @qtzar in https://github.com/avisi-cloud/structurizr-site-generatr/pull/334
* Added support for dynamic diagrams by @qtzar in https://github.com/avisi-cloud/structurizr-site-generatr/pull/333
* Add AsciiDoc support by @illnr in https://github.com/avisi-cloud/structurizr-site-generatr/pull/336
* Keep a thread safe in-memory cache of generated diagrams with links by @dirkgroot in https://github.com/avisi-cloud/structurizr-site-generatr/pull/308
* Print exception stack trace for bad branch by @pietermees in https://github.com/avisi-cloud/structurizr-site-generatr/pull/287
* Track browser package versions in package json by @jp7677 in https://github.com/avisi-cloud/structurizr-site-generatr/pull/328

### Package updates
* Update gradle/gradle-build-action action to v2.6.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/270
* Update dependency gradle to v8.2.1 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/269
* Update dependency org.jetbrains.kotlin:kotlin-scripting-jsr223 to v1.9.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/256
* Update plugin org.jetbrains.kotlin.plugin.serialization to v1.9.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/258
* Update dependency org.jetbrains.kotlinx:kotlinx-html-jvm to v0.9.1 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/271
* Update plugin org.jetbrains.kotlin.jvm to v1.9.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/257
* Update dependency net.sourceforge.plantuml:plantuml to v1.2023.10 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/273
* Update dependency com.structurizr:structurizr-dsl to v1.32.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/274
* Update gradle/gradle-build-action action to v2.7.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/275
* Update dependency com.structurizr:structurizr-core to v1.26.1 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/276
* Update dependency com.structurizr:structurizr-export to v1.16.1 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/277
* Update dependency gradle to v8.3 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/288
* Update dependency org.codehaus.groovy:groovy-jsr223 to v3.0.19 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/291
* Update dependency org.jetbrains.kotlinx:kotlinx-serialization-json to v1.6.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/292
* Update gradle/gradle-build-action action to v2.8.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/290
* Update plugin org.jetbrains.kotlin.jvm to v1.9.10 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/295
* Update dependency org.jetbrains.kotlin:kotlin-scripting-jsr223 to v1.9.10 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/293
* Update plugin org.jetbrains.kotlin.plugin.serialization to v1.9.10 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/296
* Update dependency org.eclipse.jetty.websocket:websocket-jetty-server to v11.0.16 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/297
* Update dependency org.eclipse.jetty:jetty-server to v11.0.16 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/298
* Update dependency org.eclipse.jetty:jetty-servlet to v11.0.16 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/299
* Update dependency org.eclipse.jgit:org.eclipse.jgit to v6.7.0.202309050840-r by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/302
* Update actions/checkout action to v4 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/305
* Update dependency org.jetbrains.kotlinx:kotlinx-cli to v0.3.6 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/301
* Update dependency org.slf4j:slf4j-simple to v2.0.9 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/303
* Update docker/build-push-action action to v5 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/310
* Update docker/login-action action to v3 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/311
* Update docker/setup-buildx-action action to v3 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/312
* Update dependency net.sourceforge.plantuml:plantuml to v1.2023.11 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/313
* Update dependency gradle to v8.4 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/319
* Update dependency org.eclipse.jetty:jetty-servlet to v11.0.17 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/318
* Update dependency org.eclipse.jetty.websocket:websocket-jetty-server to v11.0.17 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/316
* Update dependency org.eclipse.jetty:jetty-server to v11.0.17 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/317
* Update gradle/gradle-build-action action to v2.9.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/320
* Manually update assertk and jetty by @jp7677 in https://github.com/avisi-cloud/structurizr-site-generatr/pull/321
* Update dependency org.jsoup:jsoup to v1.16.2 by @jp7677 in https://github.com/avisi-cloud/structurizr-site-generatr/pull/330
* Update dependency org.jruby:jruby-core to v9.4.4.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/327
* Update eclipse-temurin Docker tag to v21 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/323
* Update dependency mermaid to v10.5.1 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/331
* Update dependency net.sourceforge.plantuml:plantuml to v1.2023.12 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/332
* Update dependency com.structurizr:structurizr-core to v1.27.0 by @renovate in https://github.com/avisi-cloud/structurizr-site-generatr/pull/337

## New Contributors
* @pietermees made their first contribution in https://github.com/avisi-cloud/structurizr-site-generatr/pull/287
* @illnr made their first contribution in https://github.com/avisi-cloud/structurizr-site-generatr/pull/336

**Full Changelog**: https://github.com/avisi-cloud/structurizr-site-generatr/compare/1.1.4...1.1.5

<hr /><em>This discussion was created from the release <a href='https://github.com/avisi-cloud/structurizr-site-generatr/releases/tag/1.1.5'>v1.1.5</a>.</em>

**Comment by jp7677** [comment]
cc: @qtzar @mwz

**Comment by qtzar** [comment]
@jp7677 There may be a regression bug with multi-branch support in this release.

I think it might be a result of the following change 
https://github.com/avisi-cloud/structurizr-site-generatr/commit/b867cc9d7abc7d1b64ad6f2e1c5d3ccab60c049b

Line 45 of the same file hardcodes a startpoint as "/origin/$branch" which is where the extra /origin is coming from.


```
Found the following branch: remotes/origin/MDMInterm
Found the following branch: remotes/origin/Playground
Found the following branch: remotes/origin/master
The following branches will be checked for Structurizr Workspaces: [remotes/origin/MDMInterm, remotes/origin/Playground, remotes/origin/master]
Checking branch remotes/origin/MDMInterm
Bad Branch remotes/origin/MDMInterm: Ref origin/remotes/origin/MDMInterm cannot be resolved
Checking branch remotes/origin/Playground
Bad Branch remotes/origin/Playground: Ref origin/remotes/origin/Playground cannot be resolved
Checking branch remotes/origin/master
Bad Branch remotes/origin/master: Ref origin/remotes/origin/master cannot be resolved
```

**Reply by jp7677** [reply]
Thanks for the heads-up, I've created https://github.com/avisi-cloud/structurizr-site-generatr/pull/344 which reverts the offending commit for now.

**Reply by qtzar** [reply]
I also put in a PR which will fix the issue and also fix the original problem.

**Reply by jp7677** [reply]
Yeah saw that :)
I've merged your PR and also created a new release. Hopefully all good now. Thanks a lot!

**Comment by victorlevasseur** [comment]
Thanks for the new release,
when I try to use `"generatr.site.nestGroups" "true"`, I get this error:

```
java.lang.IllegalStateException: java.lang.IllegalStateException: Property structurizr.groupSeparator not defined for model
```

If I try to add "structurizr.groupSeparator" "/" to the properties, I still get the same error. Any hint on how to fix this?

**Reply by qtzar** [reply]
Make sure you add it to the model's properties block and not the views properties block.

**Reply by jp7677** [reply]
Please see https://structurizr.com/help/properties

`structurizr.groupSeparator` needs to be defined as property directly in the DSL for the model. Structurizr itself also throws a similar message when you define nested groups. Strangely our exception appears first, I would have expected the (more verbose) one from Structurizr being thrown first when reading the DSL file ...
